its/src/test/java/com/sonar/it/scanner/msbuild/sonarcloud/IncrementalPRAnalysisSonarCloudTest.java

```diff
@@ -0,0 +1,229 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package com.sonar.it.scanner.msbuild.sonarcloud;
+
+import com.sonar.it.scanner.msbuild.utils.TestUtils;
+import com.sonar.it.scanner.msbuild.utils.VstsUtils;
+import com.sonar.orchestrator.http.HttpException;
+import com.sonar.orchestrator.util.Command;
+import com.sonar.orchestrator.util.CommandExecutor;
+import com.sonar.orchestrator.util.StreamConsumer;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class IncrementalPRAnalysisSonarCloudTest {
+  private final static Logger LOG = LoggerFactory.getLogger(IncrementalPRAnalysisSonarCloudTest.class);
+  private final static Integer COMMAND_TIMEOUT = 2 * 60 * 1000;
+  private final static String SCANNER_PATH = System.getenv("SCANNER_PATH") == null
+    ? "../build/sonarscanner-msbuild-net46/SonarScanner.MSBuild.exe" // On the local machine, the scanner is prepared by ci-build.ps1 script.
+    : System.getenv("SCANNER_PATH");
+  private final static String[] prArguments = {
+    "/d:sonar.pullrequest.base=master",
+    "/d:sonar.pullrequest.branch=pull-request-branch",
+    "/d:sonar.pullrequest.key=pull-request-key"
+  };
+  private final static String SONARCLOUD_ORGANIZATION = System.getenv("SONARCLOUD_ORGANIZATION");
+  private final static String SONARCLOUD_PROJECT_KEY = System.getenv("SONARCLOUD_PROJECT_KEY");
+  private final static String SONARCLOUD_URL = System.getenv("SONARCLOUD_URL");
+  private final static String SONARCLOUD_PROJECT_TOKEN = System.getenv("SONARCLOUD_PROJECT_TOKEN");
+
+  @ClassRule
+  public static TemporaryFolder temp = TestUtils.createTempFolder();
+
+  @Test
+  public void incrementalPrAnalysis_master() throws IOException {
```

Comment:
Just a general suggestion, maybe a bit nitpicky:
The TestClass is called `IncrementalPRAnalysis...`. I think you could rename all the `@Test` methods and remove the prefix. For example:

- `incrementalPrAnalysis_master` -> `master_EmptyCache` or `master_FullAnalysis`
- `incrementalPrAnalysis_prWithoutChanges_producesUnchangedFilesWithAllFiles` -> `prWithoutChanges_producesUnchangedFilesWithAllFiles`

Semantically you convey the same information with less repetition this way, I think.